src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -58,13 +58,20 @@ class Crawler implements \Countable, \IteratorAggregate
      */
     private bool $isHtml = true;
 
+    /**
+     * Whether the Crawler use masterminds/html5 or not.
+     */
+    private bool $isHtml5ParserUsable = true;
+
     private HTML5 $html5Parser;
 
     /**
      * @param \DOMNodeList|\DOMNode|\DOMNode[]|string|null $node A Node to use as the base for the crawling
+     * @param $isHtml5ParserUsable Set usage of masterminds/html5 or not
      */
-    public function __construct(\DOMNodeList|\DOMNode|array|string $node = null, string $uri = null, string $baseHref = null)
+    public function __construct(\DOMNodeList|\DOMNode|array|string $node = null, string $uri = null, string $baseHref = null, bool $isHtml5ParserUsable = true)
```

Comment:
`bool $useHtml5Parser = null`

when true + the mastermind is missing, we should throw a LogicException suggesting to install it (see existing similar messages in the codebase)

Reply:
Oh really ? Mastermind is required by composer.json + is actually used in `Crawler` constructor even if we dont / cant use it (e.g. `canParseHtml5String()` return false).
If we want to add this check, it's maybe better to do it even if `$useHtml5Parser` is set to `false` to avoid check in class's functions ?

Reply:
> Mastermind is required by composer.json

right, let's forget about the nullable part :)